apps/system/test/unit/task_manager_test.js

```diff
@@ -1,1377 +1,570 @@
-    setup(function() {
-      MockStackManager.mStack = [
-        appConfigs.sms,
-        appConfigs.game
-      ].map((config) => {
-        var app = new AppWindow(config);
-        apps[app.name] = app;
-        return app;
+    test('Should not show if already showing', (done) => {
+      var spyBeforeShow = spyEvent(window, 'cardviewbeforeshow');
+      assert.isTrue(tm.isShown()); // (noting that we're already shown)
+      tm.show().then(() => {
+        assert.isFalse(spyBeforeShow.called);
+        done();
       });
-      MockStackManager.mCurrent = 0;
-      MockService.mockQueryWith('AppWindowManager.getActiveWindow', apps.sms);
-
-      showTaskManager(this.sinon.clock);
     });
 
-    test('taskManager should not be active', function() {
-      taskManager.hide();
-      assert.isFalse(taskManager.isShown(), 'isShown is false');
-      assert.isFalse(cardsView.classList.contains('active'),
-                    'no .active class');
+    test('Should not respond to holdhome if already showing', () => {
+      var showSpy = sinon.spy(tm, 'show');
+      tm.respondToHierarchyEvent(new CustomEvent('holdhome'));
+      assert.isFalse(showSpy.called);
     });
 
-    test('removes classes', function() {
-      taskManager.hide();
-      assert.isFalse(screenNode.classList.contains('cards-view'));
+    test('home takes you home', () => {
+      sinon.spy(tm, 'hide');
+      tm.respondToHierarchyEvent(new CustomEvent('home'));
+      assert.ok(tm.hide.calledOnce);
     });
 
-    test('all apps should leaveTaskManager', function() {
-      MockStackManager.mStack.forEach(function(app) {
-        this.sinon.spy(app, 'leaveTaskManager');
-      }, this);
-
-      taskManager.hide();
-
-      MockStackManager.mStack.forEach(function(app) {
-        sinon.assert.calledOnce(app.leaveTaskManager);
-      }, this);
+    test('hit "home" while already closing, ignore event (bug 1203772)', () => {
+      sinon.spy(tm, 'hide');
+      tm.hide();
+      // If we receive a "home" event while already going home, we should
+      // ignore it. (i.e. in this implementation, _isTransitioning is true)
+      var result = tm.respondToHierarchyEvent(new CustomEvent('home'));
+      assert.equal(result, true);
+      assert.ok(tm.hide.calledOnce);
     });
 
-    test('hide: raises cardviewclosed event', function(done) {
-      taskManager.newStackPosition = 1;
-      waitForEvent(window, 'cardviewclosed').then(function(event) {
-        assert.equal(typeof event.detail, 'object',
-                    'gets event with detail object');
-        assert.equal(event.detail.newStackPosition, 1,
-                    'event detail reflects taskManager.newStackPosition');
-        delete taskManager.newStackPosition;
-      }, failOnReject).then(done, done);
-      taskManager.hide();
-      this.sinon.clock.tick(101);
+    test('click when empty takes you home', () => {
+      sinon.spy(tm, 'hide');
+      tm.handleEvent(new CustomEvent('click'));
```

Comment:
I usually keep variables pointing at a few of the fake dom nodes created for the test to ease this